xDripAPS.py

```diff
@@ -95,27 +163,40 @@ def get(self):
                 'unfiltered' : row[8],
                 'rssi' : row[9],
                 'noise' : row[10],
-	            'glucose' : row[4]}
+                'glucose' : row[4]}
             results_as_dict.append(result_as_dict)
 
         conn.close()
-	return results_as_dict
+        xLog("GET request returning "+ str(len(result_as_dict))+" results successfully")
+        return results_as_dict
 
     def post(self):
 
         # Get hashed API secret from request
-        request_secret_hashed = request.headers['Api_Secret']
-        print 'request_secret_hashed : ' + request_secret_hashed
-
-        # Get API_SECRET environment variable
-        env_secret_hashed = os.environ['API_SECRET']
+        try:
+            request_secret_hashed = request.headers['Api_Secret']
+            xLog('request_secret_hashed : ' + request_secret_hashed)
+        except:
+            xLog("Client didn't pass in Api-Secret header")
+            return 'Client didnt pass in Api-Secret header',500
+
+        if api_secret_xDripAPS:
+            #get API_SECRET_xDripAPS environment variable if needed
+            env_secret_hashed = os.environ['API_SECRET_xDripAPS']
+            xLog("We will authenticate using environment variable API_SECRET_xDripAPS:"+env_secret_hashed)
+       
+        elif api_secret:
+            # Get API_SECRET environment variable
+            env_secret_hashed = os.environ['API_SECRET']
+            xLog("We will authenticate using environment variable API_SECRET:"+env_secret_hashed)
 
         # Authentication check
-        if request_secret_hashed != env_secret_hashed:
```

Comment:
@ArmyAndy 
There is a bug here, you compare hash value vs. non-hash value
request_secret_hashed is really hashed, while env_secret_hashed is just the plain text value. Should be hashed before comparison